test(transformer): add tests for nested JSX `this` member expressions in arrow function transform

overlookmotel opened the pull request:
Follow-up after #5356. Handle nested JSX member expressions with `this` as base object in arrow functions transform (e.g. `<this.foo.bar />`).

overlookmotel commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5413?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#5413** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5413?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* **#5412** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5412?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#5411** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5411?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#5358** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5358?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#5357** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5357?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#5356** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5356?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @overlookmotel and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

Dunqing commented on crates/oxc_transformer/src/es2015/arrow_functions.rs:
```diff
@@ -149,20 +149,30 @@ impl<'a> Traverse<'a> for ArrowFunctions<'a> {
+                    }
+                    return;
+                }
+                JSXMemberExpressionObject::MemberExpression(member_expr) => {
+                    object = &mut member_expr.object;
+                }
```
We don't need to handle `JSXMemberExpressionObject::MemberExpression`. Our traverse will walk this. The `member_expr` is a `JSXMemberExpresion`, and its `object` is also a `JSXMemberExpressionObject` so it eventually enters `enter_jsx_member_expression_object`

overlookmotel replied:
Yes! You're absolutely right.

I've removed the changes to `enter_jsx_member_expression_object`, so this PR is now just to include tests for handling nested member expressions `<this.foo.bar>`.

overlookmotel committed:
test(transformer): add tests for nested JSX `this` member expressions in arrow function transform (#5413)

Follow-up after #5356. Handle nested JSX member expressions with `this` as base object in arrow functions transform (e.g. `<this.foo.bar />`).
